core/whistle_number_manager-1.0.0/src/wnm_number.erl

```diff
@@ -181,34 +181,29 @@ fetch_number(#number{number_db=Db
             error_number_database(Reason, N)
     end.
 
--spec maybe_correct_used_by(wnm_number()) -> wnm_number().
--spec maybe_correct_used_by(wnm_number(), wh_json:object()) -> wnm_number().
-maybe_correct_used_by(#number{assigned_to=Account}=N) ->
+
+-spec maybe_add_information(wnm_number()) -> wnm_number().
+maybe_add_information(#number{assigned_to=Account}=N) ->
     case load_phone_number_doc(Account) of
         {'ok', JObj} ->
-            maybe_correct_used_by(N, JObj);
+            maybe_add_information(N, JObj);
         {'error', _R} ->
-            lager:warning("failed to get phone number doc for correction: ~p", [_R]),
+            lager:warning("failed to get phone number doc for addition: ~p", [_R]),
             N
     end.
 
-maybe_correct_used_by(#number{number=Number
-                              ,used_by=UsedBy
+-spec maybe_add_information(wnm_number(), wh_json:object()) -> wnm_number().
+maybe_add_information(#number{number=Number
                               ,number_doc=NumberDoc
                              }=N
-                      ,JObj
-                     ) ->
-    case wh_json:get_value([Number, <<"used_by">>], JObj) of
-        UsedBy ->
-            lager:debug("~s used_by field is correct", [Number]),
-            N;
-        NewUsedBy ->
-            lager:info("correcting used_by field for number ~s from ~s to ~s"
-                      ,[Number, UsedBy, NewUsedBy]),
-            NewNumberDoc = wh_json:set_value(<<"used_by">>, NewUsedBy, NumberDoc),
-            save_number_doc(N#number{used_by=NewUsedBy
-                                     ,number_doc=NewNumberDoc
-                                    })
+                      ,JObj) ->
+    case wh_json:get_value(Number, JObj) of
+        'undefined' -> N;
+        NumberInfo ->
+            lager:info("merging doc data into number_doc for number ~s", [Number]),
+            %% This also makes sure used_by field is correct
```

Comment:
The older code was just fetching this field from `JObj` and setting it in `N`'s `number_doc` if it was different, if I'm correct.
So what this is doing now is the same behaviour, plus merging other fields. Private ones get filtered out later on.